dango/oracle/src/query.rs

```diff
@@ -43,6 +48,25 @@ fn query_price(ctx: ImmutableCtx, denom: Denom) -> anyhow::Result<PrecisionedPri
         .get_price(ctx.storage)
 }
 
+/// Does a raw query to the oracle contract to get the latest price for the given denom.
+pub fn raw_query_price(
+    querier: &QuerierWrapper,
+    denom: &Denom,
+) -> anyhow::Result<PrecisionedPrice> {
+    let oracle = querier.query_app_config(ORACLE_KEY)?;
+
+    let price = querier
+        .query_wasm_raw(oracle, PRICE_SOURCES.path(denom))?
+        .ok_or(anyhow::anyhow!(
+            "Price source not found for denom: {}",
+            denom
+        ))?
+        .deserialize_borsh::<PriceSource>()?
+        .raw_query_price(querier, oracle)?;
+
+    Ok(price)
+}
```

Comment:
i don't like this pattern: logics scattered across multiple contracts. if a query is initiated from contract X, then the query logic should be in contract X.

Reply:
Ok. I just figured probably more contracts will need the same query and then unless we want to duplicate this code everywhere we need one place to put it and I figured in the oracle itself made the most sense.

But if we make the "OracleQuerier" mentioned above then it can live there and be a separate file in the types package?